go/mysql/auth_server_static_test.go

```diff
@@ -168,6 +172,30 @@ func hupTest(t *testing.T, tmpFile *os.File, oldStr, newStr string) {
 	}
 }
 
+func hupTestWithRotation(t *testing.T, tmpFile *os.File, oldStr, newStr string) {
+	*staticFileAuthTickerDuration = time.Second * 1
+	aStatic := GetAuthServer("static").(*AuthServerStatic)
+
+	jsonConfig := fmt.Sprintf("{\"%s\":[{\"Password\":\"%s\"}]}", newStr, newStr)
+	if err := ioutil.WriteFile(tmpFile.Name(), []byte(jsonConfig), 0600); err != nil {
+		t.Fatalf("couldn't overwrite temp file: %v", err)
+	}
+
+	if aStatic.Entries[oldStr][0].Password != oldStr {
+		t.Fatalf("%s's Password should still be '%s'", oldStr, oldStr)
+	}
+
+	syscall.Kill(syscall.Getpid(), syscall.SIGHUP)
+	time.Sleep(1000 * time.Millisecond) // wait for signal handler
```

Comment:
Here, you have to wait longer because CPUs in the cloud can stall. So, let's say `20 * time.Millisecond`.